### PR TITLE
Added operationIds for OpenAPI specification

### PR DESCRIPTION
**Pull request by hoffe86**
Hey
I prepared a pull request, which add the operationId to our OpenAPI spec to improve the generation of http clients.
For example tools like [autotest](https://github.com/Azure/AutoRest) is using the OperationId as function name for the client class or low code tools like Microsoft Power Apps using the OperationId to create API facade to consume the API with their editors.

Let me know when I need to rename one or more operationId, when the name does not fit the functionality

### CODE REVIEW

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/discovery/InboxResource.java [hunk 1]
```diff
@@ -99,7 +101,7 @@ public InboxResource(final @Reference Inbox inbox) {
     @POST
     @Path("/{thingUID}/approve")
     @Consumes(MediaType.TEXT_PLAIN)
-    @Operation(summary = "Approves the discovery result by adding the thing to the registry.", responses = {
+    @Operation(operationId = "aproveInboxItemById", summary = "Approves the discovery result by adding the thing to the registry.", responses = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/discovery/InboxResource.java:104]:
```suggestion
    @Operation(operationId = "approveInboxItemById", summary = "Approves the discovery result by adding the thing to the registry.", responses = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java [hunk 2]
```diff
@@ -395,7 +395,7 @@ public Response addMember(@PathParam("itemName") @Parameter(description = "item
     @DELETE
     @RolesAllowed({ Role.ADMIN })
     @Path("/{itemName: [a-zA-Z_0-9]+}/members/{memberItemName: [a-zA-Z_0-9]+}")
-    @Operation(summary = "Removes an existing member from a group item.", security = {
+    @Operation(operationId = "removeMemberToGroupItem", summary = "Removes an existing member from a group item.", security = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java:398]:
```suggestion
    @Operation(operationId = "removeMemberFromGroupItem", summary = "Removes an existing member from a group item.", security = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java [hunk 3]
```diff
@@ -636,7 +636,7 @@ public Response createOrUpdateItem(final @Context UriInfo uriInfo, final @Contex
     @PUT
     @RolesAllowed({ Role.ADMIN })
     @Consumes(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Adds a list of items to the registry or updates the existing items.", security = {
+    @Operation(operationId =  "addsOrUpdatesItemsInRegistry", summary = "Adds a list of items to the registry or updates the existing items.", security = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java:639]:
```suggestion
    @Operation(operationId =  "addOrUpdateItemsInRegistry", summary = "Adds a list of items to the registry or updates the existing items.", security = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/link/ItemChannelLinkResource.java [hunk 4]
```diff
@@ -133,7 +133,7 @@ public Response getLink(@PathParam("itemName") @Parameter(description = "itemNam
     @PUT
     @RolesAllowed({ Role.ADMIN })
     @Path("/{itemName}/{channelUID}")
-    @Operation(summary = "Links item to a channel.", security = {
+    @Operation(operationId = "linksItemToChannel", summary = "Links item to a channel.", security = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/link/ItemChannelLinkResource.java:136]:
```suggestion
    @Operation(operationId = "linkItemToChannel", summary = "Links item to a channel.", security = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java [hunk 5]
```diff
@@ -215,7 +215,7 @@ public ThingResource( //
     @POST
     @RolesAllowed({ Role.ADMIN })
     @Consumes(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Creates a new thing and adds it to the registry.", security = {
+    @Operation(operationId = "CreateThingInRegistry", summary = "Creates a new thing and adds it to the registry.", security = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java:218]:
```suggestion
    @Operation(operationId = "createThingInRegistry", summary = "Creates a new thing and adds it to the registry.", security = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java [hunk 6]
```diff
@@ -294,7 +294,7 @@ public Response create(
     @GET
     @RolesAllowed({ Role.ADMIN })
     @Produces(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Get all available things.", responses = {
+    @Operation(operationId = "GetThings", summary = "Get all available things.", responses = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java:297]:
```suggestion
    @Operation(operationId = "getThings", summary = "Get all available things.", responses = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java [hunk 7]
```diff
@@ -314,7 +314,7 @@ public Response getAll(
     @RolesAllowed({ Role.ADMIN })
     @Path("/{thingUID}")
     @Produces(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Gets thing by UID.", security = {
+    @Operation(operationId = "GetThingById", summary = "Gets thing by UID.", security = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java:317]:
```suggestion
    @Operation(operationId = "getThingById", summary = "Gets thing by UID.", security = {
```

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java [hunk 8]
```diff
@@ -661,7 +661,7 @@ public Response getFirmwareStatus(
     @RolesAllowed({ Role.ADMIN })
     @Path("/{thingUID}/firmwares")
     @Produces(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Get all available firmwares for provided thing UID", security = {
+    @Operation(operationId = "getAvailableForThing", summary = "Get all available firmwares for provided thing UID", security = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java:664]:
```suggestion
    @Operation(operationId = "getAvailableFirmwaresForThing", summary = "Get all available firmwares for provided thing UID", security = {
```

File: bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java [hunk 9]
```diff
@@ -233,7 +233,7 @@ public Response getSitemaps() {
     @GET
     @Path("/{sitemapname: [a-zA-Z_0-9]+}")
     @Produces(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Get sitemap by name.", responses = {
+    @Operation(operationId = "getSitemapsByName", summary = "Get sitemap by name.", responses = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java:236]:
```suggestion
    @Operation(operationId = "getSitemapByName", summary = "Get sitemap by name.", responses = {
```

File: bundles/org.openhab.core.io.rest.ui/src/main/java/org/openhab/core/io/rest/ui/internal/UIResource.java [hunk 10]
```diff
@@ -104,7 +104,7 @@ public Response getAll() {
     @GET
     @Path("/components/{namespace}")
     @Produces({ MediaType.APPLICATION_JSON })
-    @Operation(summary = "Get all registered UI components in the specified namespace.", responses = {
+    @Operation(operationId = "getRegisterUIComponentsInNamespace", summary = "Get all registered UI components in the specified namespace.", responses = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.ui/src/main/java/org/openhab/core/io/rest/ui/internal/UIResource.java:107]:
```suggestion
    @Operation(operationId = "getRegisteredUIComponentsInNamespace", summary = "Get all registered UI components in the specified namespace.", responses = {
```

File: bundles/org.openhab.core.io.rest.voice/src/main/java/org/openhab/core/io/rest/voice/internal/VoiceResource.java [hunk 11]
```diff
@@ -151,7 +151,7 @@ public Response interpret(
     @POST
     @Path("/interpreters")
     @Consumes(MediaType.TEXT_PLAIN)
-    @Operation(summary = "Sends a text to the default human language interpreter.", responses = {
+    @Operation(operationId = "interpretTextByDefaultIntpreter", summary = "Sends a text to the default human language interpreter.", responses = {
```
Comment (wborn) [inline bundles/org.openhab.core.io.rest.voice/src/main/java/org/openhab/core/io/rest/voice/internal/VoiceResource.java:154]:
```suggestion
    @Operation(operationId = "interpretTextByDefaultInterpreter", summary = "Sends a text to the default human language interpreter.", responses = {
```

**Review by wborn — CHANGES_REQUESTED** [review]
In general it looks good to me. Though it would be nice if you fix the typos and make it a bit more consistent.

**Comment by hoffe86** [comment]
Hey @wborn 
I added your suggestions.
Sorry for the messy PR, there was additional signoff required :-(

**Comment by silamon** [comment]
If you don't mind, a little help to get the build green: 
* In TokenResource, there's a "(" added by accident
* In VoiceResource, you've changed a "}" to "g"
* Run mvn spotless:apply in the root folder to add or remove missing spaces where needed

**Review by wborn — CHANGES_REQUESTED** [review]
Thanks for addressing my comments!

Looks like the `operationId` is still missing on the `getApiTokens` and `removeApiToken` operations in the `TokenResource`. Can you also add it to those operations?

Will you also create a PR to add these `operationId`s to the `@Operation`s in openhab-webui?

Can you also squash all these commits into one?

**Comment by hoffe86** [comment]
Added the missing operation ids and reverted the changes for gitignore file.
@wborn  Yes I will also taking care on the operation ids in webui package

**Review by wborn — APPROVED** [review]
Thanks for adding all the IDs! :+1:

**Pull request merged**